fiftyone/core/collections.py

```diff
@@ -9155,6 +9156,35 @@ def values(
         Returns:
             the list of values
         """
+
+        # If we do not need to follow insertion order, we can potentially use a
+        # covered index query to get the values directly from the index and
+        # avoid a COLLSCAN.
+        if not _enforce_natural_order:
+            field = None
+            if isinstance(field_or_expr, str):
```

Comment:
Note that this is also a valid syntax:

```py
dataset.values("filepath", expr=F().upper())
```

so you must also ensure that `expr is None` in order to use the index pathway.

Reply:
ahhh so much packed into every function! good catch thanks.